hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -307,6 +308,143 @@ restartedNodeCanAbort tracer workDir cardanoNode hydraScriptsTxId = do
+
+      waitFor hydraTracer (20 * blockTime) [n1, n2] $
+        output "HeadIsOpen" ["utxo" .= object mempty, "headId" .= headId]
+
+      resp <-
+        parseUrlThrow ("POST " <> hydraNodeBaseUrl n2 <> "/commit")
+          <&> setRequestBodyJSON commitUTxO
+            >>= httpJSON
+
+      let depositTransaction = getResponseBody resp :: Tx
+      let tx = signTx aliceCardanoSk depositTransaction
+
+      submitTx cardanoNode tx
+
+      waitFor hydraTracer 10 [n2] $
+        output "CommitApproved" ["headId" .= headId, "utxoToCommit" .= commitUTxO]
+      waitFor hydraTracer 10 [n2] $
+        output "CommitFinalized" ["headId" .= headId, "depositTxId" .= getTxId (getTxBody tx)]
+
+      getSnapshotUTxO n1 `shouldReturn` commitUTxO
+
+      let aliceAddress = mkVkAddress networkId aliceCardanoVk
+
+      decommitTx <- do
+        let (i, o) = List.head $ UTxO.pairs commitUTxO
+        either (failure . show) pure $
+          mkSimpleTx (i, o) (aliceAddress, txOutValue o) aliceCardanoSk
+
+      let decommitUTxO = utxoFromTx decommitTx
+          decommitTxId = txId decommitTx
+
+      -- Sometimes use websocket, sometimes use HTTP
+      join . generate $
+        elements
+          [ send n2 $ input "Decommit" ["decommitTx" .= decommitTx]
+          , postDecommit n1 decommitTx
+          ]
+
+      waitFor hydraTracer 10 [n1, n2] $
+        output "DecommitRequested" ["headId" .= headId, "decommitTx" .= decommitTx, "utxoToDecommit" .= decommitUTxO]
+      waitFor hydraTracer 10 [n1, n2] $
+        output "DecommitApproved" ["headId" .= headId, "decommitTxId" .= decommitTxId, "utxoToDecommit" .= decommitUTxO]
+      failAfter 10 $ waitForUTxO cardanoNode decommitUTxO
+
+      distributedUTxO <- waitForAllMatch 10 [n1, n2] $ \v -> do
+        guard $ v ^? key "tag" == Just "DecommitFinalized"
+        guard $ v ^? key "headId" == Just (toJSON headId)
+        v ^? key "distributedUTxO" . _JSON
+
+      guard $ distributedUTxO `UTxO.containsOutputs` utxoFromTx decommitTx
+
+      pure (headId, decommitUTxO)
+
+    -- Here we post a deposit while one node is down so we can test if recover works later on
+    resp2 <-
+      parseUrlThrow ("POST " <> hydraNodeBaseUrl n1 <> "/commit")
+        <&> setRequestBodyJSON commitUTxO2
+          >>= httpJSON
+
+    let depositTransaction2 = getResponseBody resp2 :: Tx
+    let tx2 = signTx bobCardanoSk depositTransaction2
+
+    submitTx cardanoNode tx2
+
+    threadDelay $ fromIntegral (deadline * 2 + 1)
+
+    bobChainConfigFromTip <-
+      chainConfigFor Bob workDir nodeSocket hydraScriptsTxId [Alice] contestationPeriod depositDeadline
+        <&> modifyConfig (\config -> config{networkId, startChainFrom = Just tip})
+
+    withTempDir "blank-state" $ \tmpDir -> do
+      void $ readCreateProcessWithExitCode (proc "cp" ["-r", workDir </> "state-2", tmpDir]) ""
+      void $ readCreateProcessWithExitCode (proc "rm" ["-rf", tmpDir </> "state-2" </> "state"]) ""
```

Comment:
minor: add helper function to prune node's local state

Reply:
wdym? hydra-node is stopped at this point, which local state are you referring to?